extensions/amp/bundle/src/main/java/com/adobe/cq/wcm/core/extensions/amp/internal/AmpModeForwardFilter.java

```diff
@@ -64,61 +66,24 @@ public void doFilter(final ServletRequest request, final ServletResponse respons
             throws IOException, ServletException {
 
         SlingHttpServletRequest slingRequest = (SlingHttpServletRequest) request;
+        AMP_MODE ampMode = getAmpMode(slingRequest);
 
-        // Read the full selector string from the request.
-        String selectors = slingRequest.getRequestPathInfo().getSelectorString();
-        if (selectors == null) {
-            selectors = "";
-        }
+        List<String> selectors = Arrays.asList(slingRequest.getRequestPathInfo().getSelectors());
 
-        boolean hasAmpSelector = selectors.contains(AMP_SELECTOR);
-
-        String ampMode = AmpUtil.getAmpMode(slingRequest);
-
-        if (hasAmpSelector && (ampMode.equals(NO_AMP) || ampMode.isEmpty())) {
-
-            // Remove the amp selector.
-            String newSelectors;
-            if (selectors.contains(DOT + AMP_SELECTOR)) {
-                newSelectors = selectors.replace(DOT + AMP_SELECTOR, "");
-            } else if (selectors.contains(AMP_SELECTOR + DOT)) {
-                newSelectors = selectors.replace(AMP_SELECTOR + DOT, "");
-            } else {
-                newSelectors = selectors.replace(AMP_SELECTOR, "");
-            }
-
-            // Apply the updated selectors.
+        if (selectors.contains(AMP_SELECTOR) || ampMode == AMP_MODE.AMP_ONLY) {
             RequestDispatcherOptions options = new RequestDispatcherOptions();
-            options.setReplaceSelectors(newSelectors);
+            selectors = selectors.stream().filter(e -> !e.equals(AMP_SELECTOR))
```

Comment:
I would rather we just do `selectors.remove(AMP_SELECTOR)` here

Reply:
`selectors.remove(AMP_SELECTOR)` will only remove a single element. What if the amp selector is provided more than once? We should make sure that all `amp` selectors get removed.

Reply:
Ok, then we don't need the list to be writable anyway. I'll close https://github.com/adobe/aem-core-wcm-components/pull/1127#discussion_r468535568 as well.